### PR TITLE
bpo-40050: Fix importlib._bootstrap_external

### PR DESCRIPTION
**Pull request by vstinner**
Remove two unused imports: _thread and _weakref. Avoid creating a new
winreg builtin module if it's already available in sys.modules.

The winreg module is now stored as "winreg" rather than "_winreg".

https://bugs.python.org/issue40050

### CODE REVIEW

**Comment by vstinner** [comment]
I tested manually that test_importlib no longer leaks with this PR if I reapply commit 8334f30a74abcf7e469b901afc307887aa85a888:

```
$ ./python -m test -R 3:3 test_importlib 
(...)
beginning 6 repetitions
123456
......
(...)
Tests result: SUCCESS

$ git log

commit b7acc84ec0ba3dd092b0a5f879559fc1df0b5d7b (HEAD -> importlib_builtins)
Author: Hai Shi <shihai1992@gmail.com>
Date:   Fri Mar 20 16:16:45 2020 +0800

    bpo-1635741: Port _weakref extension module to multiphase initialization (PEP 489) (GH-19084)

commit ee7422bfa01164e4077140333eabe663b6aab5e9 (origin/importlib_builtins)
Author: Victor Stinner <vstinner@python.org>
Date:   Tue Mar 24 15:54:39 2020 +0100

    bpo-40050: Fix importlib._bootstrap_external
    
    Remove two unused imports: _thread and _weakref. Avoid creating a new
    winreg builtin module if it's already available in sys.modules.
    
    The winreg module is now stored as "winreg" rather than "_winreg".

(...)
```

File: Lib/importlib/_bootstrap_external.py [hunk 1]
```diff
@@ -1610,23 +1603,22 @@ def _setup(_bootstrap_module):
...
-    thread_module = _bootstrap._builtin_from_name('_thread')
-    setattr(self_module, '_thread', thread_module)
-
-    # Directly load the _weakref module (needed during bootstrap).
-    weakref_module = _bootstrap._builtin_from_name('_weakref')
-    setattr(self_module, '_weakref', weakref_module)
-
-    # Directly load the winreg module (needed during bootstrap).
+    # Directly load built-in modules needed during bootstrap.
+    builtin_names = ('_io', '_warnings', 'marshal')
```
Comment (corona10) [inline Lib/importlib/_bootstrap_external.py]: Should be ['_io', '_warnings', 'marshal']

Comment (corona10) [inline Lib/importlib/_bootstrap_external.py]:
```suggestion
    builtin_names = ['_io', '_warnings', 'marshal']
```

Comment (vstinner) [reply Lib/importlib/_bootstrap_external.py]: Yeah, I also saw this bug. I made this change locally but I lost it. I'm not sure how. Anyway, I did again the same change.

**Review by corona10 — APPROVED** [review]

File: Misc/NEWS.d/next/Library/2020-03-24-16-17-20.bpo-40050.6GrOlz.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+Fix ``importlib._bootstrap_external``. Remove two unused imports importlib:
```
Comment (brettcannon) [inline Misc/NEWS.d/next/Library/2020-03-24-16-17-20.bpo-40050.6GrOlz.rst:1]: @vstinner FYI the imports are still occurring at https://github.com/python/cpython/blob/83d46e0622d2efdf5f3bf8bf8904d0dcb55fc322/Lib/importlib/_bootstrap.py#L1168.

Comment (vstinner) [reply Misc/NEWS.d/next/Library/2020-03-24-16-17-20.bpo-40050.6GrOlz.rst:1]: Right, I know. That's why I wrote "Fix importlib._bootstrap_external."

Do you think that the NEWS entry is misleading? If yes, can you suggest a better phrasing?

Comment (brettcannon) [reply Misc/NEWS.d/next/Library/2020-03-24-16-17-20.bpo-40050.6GrOlz.rst:1]: I would go with:
```
Remove redundant imports in ``importlib._bootstrap_external``.
```

Comment (vstinner) [reply Misc/NEWS.d/next/Library/2020-03-24-16-17-20.bpo-40050.6GrOlz.rst:1]: I wrote PR #19148 to change the NEWS entry.

**Pull request merged**